src/renderers/native/ReactNativeFiberInspector.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactNativeFiberInspector
+ * @flow
+ */
+'use strict';
+
+const ReactNativeComponentTree = require('ReactNativeComponentTree');
+const ReactFiberTreeReflection = require('ReactFiberTreeReflection');
+const getComponentName = require('getComponentName');
+
+import type {Fiber} from 'ReactFiber';
+
+if (__DEV__) {
+  var traverseOwnerTreeUp = function(hierarchy, instance: any) {
+    if (instance) {
+      hierarchy.unshift(instance);
+      traverseOwnerTreeUp(hierarchy, instance._debugOwner);
+    }
+  };
+
+  var getOwnerHierarchy = function(instance: any) {
+    var hierarchy = [];
+    traverseOwnerTreeUp(hierarchy, instance);
+    return hierarchy;
+  };
+
+  var lastNotNativeInstance = function(hierarchy) {
+    for (let i = hierarchy.length - 1; i > 1; i--) {
+      const instance = hierarchy[i];
+      if (!instance.viewConfig) {
+        return instance;
+      }
+    }
+    return hierarchy[0];
+  };
+
+  var getHostNode = function(fiber: Fiber, findNodeHandle) {
+    let hostNode;
+    // look for children first for the hostNode
+    // as composite fibers do not have a hostNode
+    while (fiber) {
+      hostNode = findNodeHandle(fiber.stateNode);
```

Comment:
Can we first check that fiber type is either `ClassComponent` or `HostComponent`? I don't think it's universally safe to take `.stateNode` and expect it to be either a class or a node. For example coroutines also use `.stateNode` for a different purpose, and `findNodeHandle()` would blow up on unexpected inputs.

I would imagine something like:

```js
if (fiber.stateNode !== null && (fiber.type === HostComponent || fiber.type === ClassComponent)) {
  hostNode = findNodeHandle(fiber.stateNode);
}
```

I’m actually not sure if we even need to check for `ClassComponent` there because we’d drill down anyway. So maybe it’s enough to check `fiber.type === HostComponent and fiber.stateNode !== null`.

WDYT?